packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -130,6 +136,23 @@ export async function transformWithEsbuild(
     tsconfigRaw
   } as ESBuildOptions
 
+  if (
```

Comment:
```suggestion
  // esbuild uses tsconfig fields when both the normal options and tsconfig was set
  // but we want to prioritize the normal options
  if (
```
It would be nice to have a comment here, I think.